### PR TITLE
doc(Changelogs): Add Changelog for 1.0.0 and prepare docs for release

### PR DESCRIPTION
**Pull request by kgriffs**

### CODE REVIEW

**Comment by jmvrbanac** [comment]
:+1: Lots of changes this round!

File: doc/changes/0.2.0.rst [hunk 1]
```diff
@@ -1,8 +1,23 @@
 Changelog for Falcon 0.2.0
 ==========================
 
-New
----
+Breaking Changes
```
Comment (kgriffs) [inline doc/changes/0.2.0.rst]: Just reordered the sections to be consistent with the 1.0.0 release notes. I wanted to make sure people didn't gloss over the breaking changes and miss something that affects them.

File: doc/changes/1.0.0.rst [hunk 2]
```diff
@@ -0,0 +1,147 @@
+Changelog for Falcon 1.0.0
+==========================
+
+Breaking Changes
+----------------
+- The deprecated global hooks feature has been removed.
+  :class:`~falcon.API` no longer accepts `before` and `after`
```
Comment (kgriffs) [inline doc/changes/1.0.0.rst]: Getting fancy with sphinx directives FTW!

File: doc/changes/0.2.0.rst (hunk 1 above)
Comment (fxfitz) [inline doc/changes/0.2.0.rst]: +1

**Comment by fxfitz** [comment]
LGTM as well!

**Comment by stevelle** [comment]
looks good at a quick review, all comments read well and references seem correct

**Pull request merged**